python/cudf/cudf/utils/utils.py

```diff
@@ -532,3 +537,18 @@ def get_relevant_submodule(func, module):
         else:
             return None
     return module
+
+
+def create_cat_column_from_pd_cat_scalar(cat_scalar, size):
```

Comment:
Here, could `cat_scalar` also be a cuDF categorical? If so, suggest renaming to:

```suggestion
_categorical_scalar_broadcast_to(cat_scalar, size):
```

that way we don't scare people (i.e., us, in the future) away from using it to broadcast cuDF categoricals.

Reply:
@shwina . Currently, it can't be a `cudf.Series`  because the function relies on accessing  `cat_scalar.categories` and `cat_scalar.codes`, though it will work if we provide it with the category accessor.  

That is below works 
```python
cat_s = cudf.Series(pd.Categorical.from_codes(np.zeros(2, dtype=int), categories=['cat.csv']))

output_col = create_cat_column_from_pd_cat_scalar(cat_s.cat,2)
cudf.Series(output_col)
```
```python
0    cat.csv
1    cat.csv
dtype: category
Categories (1, object): ['cat.csv']
```

But below fails:
```python
output_col = create_cat_column_from_pd_cat_scalar(cat_s,2)
cudf.Series(output_col)
```

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-0a88b3ae8589> in <module>
      1 cat_s = cudf.Series(pd.Categorical.from_codes(np.zeros(2, dtype=int), categories=['cat.csv']))
      2 
----> 3 output_col = create_cat_column_from_pd_cat_scalar(cat_s,2)
      4 cudf.Series(output_col)

/nvme/0/vjawa/vjawa_cudf/cudf/python/cudf/cudf/utils/utils.py in create_cat_column_from_pd_cat_scalar(cat_scalar, size)
    542 def create_cat_column_from_pd_cat_scalar(cat_scalar, size):
    543 
--> 544     cats = column.as_column(cat_scalar.categories)
    545     codes = scalar_broadcast_to(cat_scalar.codes[0], size)
    546 

AttributeError: 'Series' object has no attribute 'categories'
```

I am not sure what you meant by `a cuDF categorical`. We can modify this function to check if the input is a  `cudf.Series`  and then get cats and codes accordingly. Please let me know if that's what you have in mind.

Reply:
Sorry - I was mistaken :) I thought we had a `cudf.Categorical` class similar to `pd.Categorical`, but we don't.

I'd still lean towards something like `_categorical_scalar_broadcast_to()` as far as naming is concerned. Underscore because we don't want people using this function outside this module (just use `scalar_broadcast_to`) instead.

Reply:
@shwina , Gotcha. I agree with that and also having a general util is better so with commit https://github.com/rapidsai/cudf/pull/6830/commits/b479d60762c81d1ffbe6723bd7d1910834318f5a we can now handle all of the below cases:

```python
cat_ar = pd.Categorical.from_codes(np.zeros(1, dtype=int), categories=['cat.csv'])

## pd.Categorical ar
o = _categorical_scalar_broadcast_to(cat_ar,2)
print(cudf.Series(o))

## pandas Categorical series
o = _categorical_scalar_broadcast_to(pd.Series(cat_ar),2)
print(cudf.Series(o))

## cudf Categorical Series
o = _categorical_scalar_broadcast_to(cudf.Series(cat_ar),2)
print(cudf.Series(o))

### cudf.Categorical column
o = _categorical_scalar_broadcast_to(cudf.Series(cat_ar)._column,2)
print(cudf.Series(o))
```

```
0    cat.csv
1    cat.csv
dtype: category
Categories (1, object): ['cat.csv']
0    cat.csv
1    cat.csv
dtype: category
Categories (1, object): ['cat.csv']
0    cat.csv
1    cat.csv
dtype: category
Categories (1, object): ['cat.csv']
0    cat.csv
1    cat.csv
dtype: category
Categories (1, object): ['cat.csv']

```

Reply:
Let me know if there is anything else I need to address.